modules/oe_paragraphs_timeline/oe_paragraphs_timeline.module

```diff
@@ -23,6 +24,95 @@ function oe_paragraphs_timeline_field_widget_timeline_widget_form_alter(&$elemen
     return;
   }
 
-  $element['label']['#required'] = TRUE;
-  $element['title']['#required'] = TRUE;
+  // We can't use the protected isDefaultValueWidget() method.
+  $is_default_value_widget = (bool) $form_state->get('default_value_widget');
+  if ($is_default_value_widget) {
+    return;
+  }
+
+  $element['label']['#required'] = $element['#required'];
+  $element['title']['#required'] = $element['#required'];
+  $element['#element_validate'][] = '_oe_paragraphs_validate_required_fields';
+
+  // If the elements were set as required, there is no need to append states.
+  if ($element['#required']) {
+    return;
+  }
+
+  // Get the selector to build the paths for each field.
+  $field_name = $field_definition->getName();
+  $delta = $element['#delta'];
+  $parents = $element['#field_parents'];
+  $parents[] = $field_name;
+  $parents[] = $delta;
+  $selector = array_shift($parents);
+  if ($parents) {
+    $selector .= '[' . implode('][', $parents) . ']';
+  }
+
+  // Mark label as required when title or content are filled in.
+  $element['label']['#states']['required'][] = [
+    'textarea[name="' . $selector . '[body][value]"]' => ['filled' => TRUE],
+  ];
+  $element['label']['#states']['required'][] = [
+    ':input[name="' . $selector . '[title]"]' => ['filled' => TRUE],
+  ];
```

Comment:
Theoretically it is possible to use 'or' inside form states: build/core/modules/system/tests/modules/form_test/src/Form/JavascriptStatesForm.php 
`    $form['textfield'] = [
      '#type' => 'textfield',
      '#title' => 'textfield',
      '#states' => [
        'visible' => [
          [':input[name="select"]' => ['value' => '1']],
          'or',
          [':input[name="number"]' => ['value' => '1']],
        ],
      ],
    ];`

Reply:
I checked and there no 'or' support, only xor. Or is implied by having different arrays.

Reply:
ok I see